library/parse_backup_metadata.py

```diff
@@ -46,6 +46,8 @@ def get_puppet_version(puppet_agent_version, puppet_rpm_version):
            return "4"
         elif puppet_agent_version == "5":
            return "5"
+        elif puppet_agent_version == "6":
+           return "6"
         else:
             raise error_msg
```

Comment:
This will not work and raise a `TypeError: exceptions must be old-style classes or derived from BaseException, not str`, same below in line 55. You could `raise Exception(error_msg)` instead.

Or we drop support for Satellite <6.4 and this function becomes:

```python
def get_puppet_version(puppet_agent_version):
    if puppet_agent_version == "1":
        return "4"
    else:
        return puppet_agent_version
```

(and `parse_backup_metadata` is changed to not look at the "puppet" RPM at all)

Reply:
And if we go 6.5+, we don't even need that, as 6.5 had Puppet5, so this whole function is not needed anymore :)

Reply:
+1 to dropping support for 6.4 and older and dropping this function entirely

Reply:
@evgeni Thank you for the review. I will work on this during this weekend and I will incorporate those changes too.

Reply:
@mccun934 @evgeni 

Thank you for the review. Based on the above discussion, We agreed to drop support for 6.4 and older version. So removed function completely.